zerver/lib/export.py

```diff
@@ -1038,6 +1038,10 @@ def export_files_from_s3(realm: Realm, bucket_name: str, output_dir: Path,
             record['realm_id'] = user_profile.realm_id
         record['user_profile_email'] = user_profile.email
 
+        # Fix the record ids
+        record['user_profile_id'] = int(record['user_profile_id'])
+        record['realm_id'] = int(record['realm_id'])
+
```

Comment:
As before, it would be good to mention in the commit if you found this from the tests, or just noticed it and found a fix.

Since this is a type issue, this suggests to me that a `TypedDict` might be even more useful to ensure this doesn't break accidentally again. That would be a static test for this issue (via mypy) or else perhaps we need a regular test for this?

Reply:
I'll update this with tests.

Reply:
The other tests for this are included in 6ddca60b16f59e16f102ed810ccf711fe5dabf7c (Where these if a possibilty of any breakage)